File: modules/cubex-core/src/main/java/org/cubexmc/core/CubexLogger.java
package org.cubexmc.core;

import java.util.Objects;
import java.util.logging.Level;
import java.util.logging.Logger;

public final class CubexLogger {

    private final Logger delegate;

    public CubexLogger(Logger delegate) {
        this.delegate = Objects.requireNonNull(delegate, "delegate");
    }

    public void info(String message) {
        delegate.info(message);
    }

    public void warn(String message) {
        delegate.warning(message);
    }

    public void warn(String message, Throwable throwable) {
        log(Level.WARNING, message, throwable);
    }

    public void severe(String message) {
        delegate.severe(message);
    }

    public void severe(String message, Throwable throwable) {
        log(Level.SEVERE, message, throwable);
    }

    public void debug(String message) {
        delegate.fine(message);
    }

    public void log(Level level, String message, Throwable throwable) {
        delegate.log(level, message, throwable);
    }
}

File: modules/cubex-core/src/main/kotlin/org/cubexmc/core/CubexLogger.kt
package org.cubexmc.core

import java.util.Objects
import java.util.logging.Level
import java.util.logging.Logger

class CubexLogger(delegate: Logger) {
    private val delegate: Logger = Objects.requireNonNull(delegate, "delegate")

    fun info(message: String) {
        delegate.info(message)
    }

    fun warn(message: String) {
        delegate.warning(message)
    }

    fun warn(message: String, throwable: Throwable) {
        log(Level.WARNING, message, throwable)
    }

    fun severe(message: String) {
        delegate.severe(message)
    }

    fun severe(message: String, throwable: Throwable) {
        log(Level.SEVERE, message, throwable)
    }

    fun debug(message: String) {
        delegate.fine(message)
    }

    fun log(level: Level, message: String, throwable: Throwable) {
        delegate.log(level, message, throwable)
    }
}
